src/Microsoft.Identity.Client/UiRequiredExceptionClassification.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Details about the cause of an <see cref="MsalUiRequiredException"/>, giving a hint about what the user can expect when 
+    /// they go through interactive authentication. See https://aka.ms/msal-net-UiRequiredException for details.
+    /// </summary>
+    public enum UiRequiredExceptionClassification
+    {
+        /// <summary>
+        /// No further details are provided. It is possible that the user will be able to resolve the issue by launching interactive authentication.
+        /// See https://aka.ms/msal-net-UiRequiredExceptionfor details
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// Condition cannot be resolved at this time. Launching interactive authentication flow will show a message explaining the condition.
+        /// See https://aka.ms/msal-net-UiRequiredExceptionfor details
+        /// </summary>
+        MessageOnly,
+
+        /// <summary>
+        /// Condition can be resolved by user interaction during the interactive authentication flow.
+        /// See https://aka.ms/msal-net-UiRequiredExceptionfor details
+        /// </summary>
+        BasicAction,
+
+        /// <summary>
+        /// Condition can be resolved by additional remedial interaction with the system, outside of the interactive authentication flow.
+        /// See https://aka.ms/msal-net-UiRequiredExceptionfor details
+        /// </summary>
+        AdditionalAction,
```

Comment:
not a ui exception classification

Reply:
I disagree, UI is required to display this "Additional action". However, developers can give a visual hint to the user that they won't be able to login to the app anytime soon :). 

From the spec: 

_Call AcquireTokenInteractively() to show a message that explains the remedial action. Calling application may choose to hide flows that require additional_action if the user is unlikely to complete the remedial action._

Reply:
Updated the comments to better reflect this.

Reply:
IMO: not the intention of the UIRequiredException. Suppose it will work regardless and developers will get used to this.

---
In reply to: [300998582](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1254#discussion_r300998582) [](ancestors = 300998582)